flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/MergeTreeWriter.java

```diff
@@ -111,7 +114,10 @@ public void write(ValueKind valueKind, RowData key, RowData value) throws Except
 
     private void flush() throws Exception {
         if (memTable.size() > 0) {
-            finishCompaction();
+            if (levels.numberOfSortedRuns() > numSortedRunStopTrigger) {
```

Comment:
Currently number of sorted runs is limited by `compaction due to file num`. If we add async compaction it is very likely that number of sorted runs exceed the limit. What to do under this scenario?

Reply:
In fact the biggest benefit is the decoupling of the compactionTrigger parameter and the stopTrigger parameter, giving the user the flexibility to define the behavior.